### PR TITLE
build person profile

### PR DESCRIPTION
**Pull request by icerove**
add people.set function to the eventtracking.js to build user profile

### CODE REVIEW

File: utils/eventtracking.js [hunk 1]
```diff
@@ -40,41 +47,53 @@ const track = async (eventType, eventProperties, options) => {
...
                 rl.question(
                     chalk`We would like to collect data on near-cli usage to improve developer experience.` +
-                    chalk` We will never send private information. We only collect which commands are run via an anonymous identifier.` +
-                    chalk`{bold.yellow  Would you like to opt in (y/n)? }`,
+                        chalk` We will never send private information. We only collect which commands are run via an anonymous identifier.` +
+                        chalk`{bold.yellow  Would you like to opt in (y/n)? }`,
                     async (consentToEventTracking) => {
-                        if (consentToEventTracking == 'y' || consentToEventTracking == 'Y') {
+                        if (
+                            consentToEventTracking == 'y' ||
```
Comment (vgrichina) [inline utils/eventtracking.js:84]: nit: just use `.toLowerCase()` on `consentToEventTracking`? Also why is this touched at all?

File: utils/eventtracking.js [hunk 2]
```diff
@@ -90,11 +109,23 @@ const askForConsentIfNeeded = async (options) => {
     const shellSettings = settings.getShellSettings();
     // if the appropriate option is not in settings, ask now and save settings.
     if (!(TRACKING_ENABLED_KEY in shellSettings)) {
-        shellSettings[TRACKING_ENABLED_KEY] = shouldOptInByDefault() || await getEventTrackingConsent();
-        shellSettings[TRACKING_SESSION_ID_KEY] = shellSettings[TRACKING_ENABLED_KEY] ? uuid.v4() : undefined;
+        shellSettings[TRACKING_ENABLED_KEY] =
```
Comment (vgrichina) [inline utils/eventtracking.js:112]: is this intentional change of formatting?

File: utils/eventtracking.js [hunk 3]
```diff
@@ -90,11 +109,23 @@ const askForConsentIfNeeded = async (options) => {
...
+        shellSettings[TRACKING_SESSION_ID_KEY] = shellSettings[
+            TRACKING_ENABLED_KEY
+        ]
+            ? uuid.v4()
+            : undefined;
         settings.saveShellSettings(shellSettings);
         if (shellSettings[TRACKING_ENABLED_KEY]) {
-            track(module.exports.EVENT_ID_TRACKING_OPT_IN, {}, options);
+            await track(module.exports.EVENT_ID_TRACKING_OPT_IN, {}, options);
+            await mixpanel.people.set({
```
Comment (vgrichina) [inline utils/eventtracking.js:122]: how does this work? If this is set globally, why is it also sent with `track`? Why call `mixpanel.people.set` after `track` and not before?

Comment (icerove) [reply utils/eventtracking.js:122]: track function is for event tracking, and people.set function is setting user profile, it does not matter for the sequence in my mind. Is that matter? @vgrichina

File: utils/eventtracking.js (hunk 2 above)
Comment (icerove) [reply utils/eventtracking.js:112]: I think so, this is all after I command `yarn fix`

File: utils/eventtracking.js (hunk 1 above)
Comment (icerove) [reply utils/eventtracking.js:84]: I do not touch this, I think it is from eslint fixing

**Comment by icerove** [comment]
Is this okay to merge ? @vgrichina

File: utils/eventtracking.js [hunk 4]
```diff
@@ -40,41 +47,49 @@ const track = async (eventType, eventProperties, options) => {
...
-                    chalk`{bold.yellow  Would you like to opt in (y/n)? }`,
+                        chalk` We will never send private information. We only collect which commands are run via an anonymous identifier.` +
+                        chalk`{bold.yellow  Would you like to opt in (y/n)? }`,
                     async (consentToEventTracking) => {
-                        if (consentToEventTracking == 'y' || consentToEventTracking == 'Y') {
+                        if (consentToEventTracking.toLowerCase() == 'y') {
                             resolve(true);
-                        } else if (consentToEventTracking == 'n' || consentToEventTracking == 'N') {
+                        } else if (
+                            consentToEventTracking.toLowerCase() == 'n'
```
Comment (vgrichina) [inline utils/eventtracking.js:86]: nit: formatting is a bit weird (no reason not to keep `if` and condition on same line)

**Comment by vgrichina** [comment]
@icerove merged, thanks! Let me know if we need to cut a release

**Pull request merged**